### PR TITLE
[DW-1379] Added announcement doctype

### PR DESCRIPTION
**Pull request by alexandru-mitcu-valtech**
[DW-1379] Added Java bean

[DW-1379] Converted expiryDate to datetime

[DW-1379] Added Announcements main folder

[DW-1379] Added new-intranet-document query
[DW-1379] Added BR config and freemarker template for Announcements

[DW-1379] Restricted allowed doctypes in intranet

### CODE REVIEW

File: site/components/src/main/java/uk/nhs/digital/intranet/components/AnnouncementComponent.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    @Override
+    public void doBeforeRender(HstRequest request, HstResponse response) throws HstComponentException {
+        super.doBeforeRender(request, response);
+        HstRequestContext requestContext = RequestContextProvider.get();
+        HippoBean contentBean = requestContext.getContentBean();
+
+        if (contentBean != null) {
+            request.setAttribute("document", contentBean);
+        }
```
Comment (mohammed-hasan-valtech) [inline site/components/src/main/java/uk/nhs/digital/intranet/components/AnnouncementComponent.java:21]: Can you just use the BaseGaComponent ? Does the same thing + also allows full ga tracking in html fields

Comment (mohammed-hasan-valtech) [inline site/components/src/main/java/uk/nhs/digital/intranet/components/AnnouncementComponent.java:21]: Can you just use the BaseGaComponent ? Does the same thing + also allows full ga tracking in html fields

Comment (alexandru-mitcu-valtech) [reply site/components/src/main/java/uk/nhs/digital/intranet/components/AnnouncementComponent.java:21]: Good spot

File: site/components/src/main/java/uk/nhs/digital/intranet/components/AnnouncementComponent.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package uk.nhs.digital.intranet.components;
+
+import org.hippoecm.hst.core.component.HstComponentException;
+import org.hippoecm.hst.core.component.HstRequest;
+import org.hippoecm.hst.core.component.HstResponse;
+import uk.nhs.digital.common.components.BaseGaContentComponent;
+
+public class AnnouncementComponent extends BaseGaContentComponent {
```
Comment (mohammed-hasan-valtech) [inline site/components/src/main/java/uk/nhs/digital/intranet/components/AnnouncementComponent.java:8]: Can you directly use BaseGaContentComponent in the hst:page? it removes the extra component being instantiated

Comment (alexandru-mitcu-valtech) [reply site/components/src/main/java/uk/nhs/digital/intranet/components/AnnouncementComponent.java:8]: Oooh, that's what you meant. It shouldn't be any problem to re-use the same Java component for multiple hst:pages, correct?

Comment (alexandru-mitcu-valtech) [reply site/components/src/main/java/uk/nhs/digital/intranet/components/AnnouncementComponent.java:8]: On the other hand, I would argue against this on the basis that this is not something done in the codebase (or at least I didn't notice it). My view on this is that every page should have its own component, and that the `BaseGaContentComponent` should only be extended, not directly used. The same as with `BaseHstDocument`.

Comment (mohammed-hasan-valtech) [reply site/components/src/main/java/uk/nhs/digital/intranet/components/AnnouncementComponent.java:8]: I believe it is recommended to reuse these base components for multiple simple doctypes, but its not a big deal either way, I'll approve.

**Pull request merged**